app/src/main/java/com/sopt/now/feature/auth/LoginViewModel.kt

```diff
@@ -20,76 +21,60 @@ import kotlinx.coroutines.launch
 
 @HiltViewModel
 class LoginViewModel @Inject constructor(
-    private val getUserInfoUseCase: GetUserInfoUseCase,
-    private val getCheckLoginUseCase: GetCheckLoginUseCase,
-    private val saveCheckLoginUseCase: SaveCheckLoginUseCase
+    private val getUserIdUseCase: GetUserIdUseCase,
+    private val saveUserIdUseCase: SaveUserIdUseCase,
+    private val loginRepository: LoginRepository
 ) : ViewModel() {
     private val _savedUserInfo = MutableStateFlow<UserEntity>(
         UserEntity(
             id = "",
             password = "",
             nickName = "",
-            mbti = ""
+            phone = ""
         )
     )
-    val savedUserInfo: StateFlow<UserEntity> get() = _savedUserInfo
-
-    private val _loginState = MutableSharedFlow<UiState<UserEntity>>()
-    val loginState: SharedFlow<UiState<UserEntity>> get() = _loginState.asSharedFlow()
 
     private val _autoLoginState = MutableStateFlow<Boolean>(false)
     val autoLoginState: StateFlow<Boolean> get() = _autoLoginState.asStateFlow()
 
+    private val _postLoginState = MutableSharedFlow<UiState<Int>>()
+    val postLoginState: SharedFlow<UiState<Int>> get() = _postLoginState.asSharedFlow()
+
     init {
         checkAutoLogin()
     }
 
     private fun checkAutoLogin() {
         viewModelScope.launch {
             _autoLoginState.value = isAutoLogin()
-            if (!isAutoLogin()) {
-                _savedUserInfo.value = getUserInfoUseCase.invoke()
-            }
         }
     }
 
-    private fun isAutoLogin(): Boolean = getCheckLoginUseCase.invoke()
+    private fun isAutoLogin(): Boolean = getUserIdUseCase.invoke() != HEADER_ID_DEFAULT_NUM
 
-    fun saveCheckLoginSharedPreference(input: Boolean) {
-        saveCheckLoginUseCase.invoke(input)
+    fun saveCheckLoginSharedPreference(input: Int) {
+        viewModelScope.launch {
+            saveUserIdUseCase.invoke(input)
+        }
     }
 
     fun setSavedUserInfo(input: UserEntity) {
         _savedUserInfo.value = input
     }
 
-    fun setLogin(id: String, pwd: String) {
-        checkLoginValidate(id, pwd)
-    }
-
-    private fun checkLoginValidate(id: String, pwd: String) {
+    fun postLogin(id: String, pwd: String) {
         viewModelScope.launch {
```

Comment:
[P1]
launch의 인자로 Dispatcher.IO를 전달하여 비동기적으로 수행하면 좋을 것 같습니다~
```suggestion
        viewModelScope.launch(Dispatcher.IO) {
```

Reply:
혹시 retroift2와 suspend를 같이 사용하게 되면 자체적으로 Dispatcheer.IO가 할당 되는 것으로 알고 있는데 명시 하는게 더 좋을까요??

Reply:
오호.. 그건 저도 처음 알았네요 
retrofit2 이외에 다른 처리 동작이 없는 경우에는 명시하지 않아도 될 것 같아요 :)